testsuite/MDAnalysisTests/coordinates/test_gro.py

```diff
@@ -213,6 +213,74 @@ def test_writer(self):
                             "not reproduce original coordinates")
 
     @dec.slow
+    def test_writer_no_resnames_consistency(self):
+        self.universe.residues.resnames = ''
+        self.universe.atoms.write(self.outfile)
+        u = mda.Universe(self.outfile)
+        assert_equal(u.atoms.resnames, self.universe.atoms.resnames)
+
+    @dec.slow
+    def test_writer_no_resnames_UNK(self):
+        self.universe.residues.resnames = ''
```

Comment:
So this test will fail, but the test above will pass, which I think is the desired behaviour... A round trip should work even if the resnames were empty strings?

This Universe topology has resnames, even if they are set to `''`.  Instead if you supply a Universe without resnames (eg from an xyz file), it should still write out something.  I've been using a function called `make_Universe` to create a Universe with an arbitrary set of attributes..

```python
from MDAnalysisTests.core.groupbase import make_Universe
# make a Universe without resnames
# AtomGroup.resnames won't work in this Universe
u = make_Universe(['names', 'resids'])

u.atoms.write(self.outfile)

u2 = mda.Universe(self.outfile)  # this should work, but raise a warning about the resnames
u2.atoms.resnames  # these should now exist, but be UNK
```

Sorry about not making this clear before